src/praisonai-bot/praisonai_bot/gateway/redis_pubsub.py

```
@@ -45,6 +60,49 @@ def __init__(self, config: RedisConfig) -> None:
+            from praisonaiagents.gateway.degraded_state import DegradedOwner
+
+            registry.mark(
+                DegradedOwner(
+                    owner_kind=self._DEGRADED_OWNER_KIND,
+                    owner_id=self._DEGRADED_OWNER_ID,
+                    state="stale",
+                    reason=f"redis pub/sub disconnected: {reason}",
+                    retry_hint="check Redis connectivity; see `praisonai gateway doctor`",
+                )
+            )
+        except Exception as e:  # pragma: no cover - registry is best-effort
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)